src/foam/nanos/http/LiveScriptBundler.java

```diff
@@ -64,42 +63,59 @@ public WatcherThread(
       listener_ = listener;
     }
 
-    // Standard WatchService loop
-    public void run() {
-      for ( ; ; ) {
-        WatchKey key;
-        try {
-          key = watcher_.take();
-        } catch (InterruptedException x) {
-          return;
+    public void tick() {
+      WatchKey key = watcher_.poll();
+      if ( key == null ) return;
+
+      for ( WatchEvent<?> event : key.pollEvents() ) {
+        WatchEvent.Kind<?> kind = event.kind();
+        if ( kind == OVERFLOW ) {
+          log_("ERROR", "File watch buffer overflowed!");
+          continue;
         }
-        for ( WatchEvent<?> event : key.pollEvents() ) {
-          WatchEvent.Kind<?> kind = event.kind();
-          if ( kind == OVERFLOW ) {
-            log_("ERROR", "File watch buffer overflowed!");
-            continue;
-          }
 
-          WatchEvent<Path> ev = (WatchEvent<Path>) event;
-          Path filename = ev.context();
+        WatchEvent<Path> ev = (WatchEvent<Path>) event;
+        Path filename = ev.context();
 
-          // Ex: foamPath="foam/core/Property.js"
-          String foamPath = foamDir_ + "/" + filename.toString();
+        // Ex: foamPath="foam/core/Property.js"
+        String foamPath = foamDir_ + "/" + filename.toString();
 
-          if ( fileNames_.contains(foamPath) ) {
-            log_("UPDATE", foamPath);
+        if ( fileNames_.contains(foamPath) ) {
+          log_("UPDATE", foamPath);
 
-            // Run the javascript builder
-            listener_.onFileUpdate(foamPath, realDir_.resolve(filename));
+          // Run the javascript builder
+          listener_.onFileUpdate(foamPath, realDir_.resolve(filename));
 
-          } else {
-            log_("IGNORE", foamPath);
-          }
+        } else {
+          log_("IGNORE", foamPath);
+        }
+
+        if ( ! key.reset() ) break;
+      }
+    }
+  }
 
-          if ( ! key.reset() ) break;
+  private class WatcherThread implements Runnable {
```

Comment:
If LiveScriptBundler implements NanoService, then it will call 'start' and from here start your thread or timer.  Then startup is consistent.